### ISSUE TITLE
Randomly being redirected to internal api urls

### ISSUE THREAD

**Issue by gerane** [ironmansoftware/universal-dashboard#1194]
#### Describe the Issue
With 2.6.2 I am on occasion randomly redirected to internal api urls. It's usually when the page is sitting idle. The last one was for an api url for a endpoint that refreshes data. The json for the endpoint is displayed on the page.

Url:
https://siteurl/api/internal/component/element/Alerts

Snippet:
```powershell
New-UDElement -Tag "div" -Id "Alerts" -AutoRefresh -RefreshInterval 30 -Endpoint {
    Blah
}
```

#### Version Information
- **Operating System**: Windows 10
- **PowerShell Version**: 5.1.18362.145
- **Universal Dashboard Version**: 2.6.2
- **UD Hosting Method**: Powershell

**Comment by gerane** [comment]
I think this might have something to do with adfs auth. On my dev box, when I restarted the dashboard, when it reloaded it redirected me to the same alerts internal api, but this time the page just said wsfed.

Here is some of the log output. I was on a page inside of the dashboard when I reloaded it.

```
8:07:50 [Info] Quartz.Impl.StdSchedulerFactory Using object serializer: Quartz.Simpl.BinaryObjectSerializer, Quartz
08:07:50 [Info] Quartz.Core.SchedulerSignalerImpl Initialized Scheduler Signaller of type: Quartz.Core.SchedulerSignalerImpl
08:07:50 [Info] Quartz.Core.QuartzScheduler Quartz Scheduler v.3.0.6.0 created.
08:07:50 [Info] Quartz.Simpl.RAMJobStore RAMJobStore initialized.
08:07:50 [Info] Quartz.Core.QuartzScheduler Scheduler meta-data: Quartz Scheduler (v3.0.6.0) 'QuartzScheduler' with instanceId 'NON_CLUSTERED'
  Scheduler class: 'Quartz.Core.QuartzScheduler' - running locally.
  NOT STARTED.
  Currently in standby mode.
  Number of jobs executed: 0
  Using thread pool 'Quartz.Simpl.DefaultThreadPool' - with 100 threads.
  Using job-store 'Quartz.Simpl.RAMJobStore' - which does not support persistence. and is not clustered.

8:07:50 [Info] Quartz.Impl.StdSchedulerFactory Quartz scheduler 'QuartzScheduler' initialized
08:07:50 [Info] Quartz.Impl.StdSchedulerFactory Quartz scheduler version: 3.0.6.0
08:07:50 [Info] Quartz.Core.QuartzScheduler Scheduler QuartzScheduler_$_NON_CLUSTERED started.
08:07:54 [Info] Microsoft.AspNetCore.Hosting.Internal.WebHost Request starting HTTP/1.1 GET https://localhost:10002/api/internal/component/element/Alerts  
08:07:54 [Info] Microsoft.AspNetCore.Authentication.Cookies.CookieAuthenticationHandler Cookies was not authenticated. Failure message: Ticket expired
08:07:54 [Info] Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker Route matched with {action = "Element", controller = "Component"}. Executing action UniversalDashboard.Controllers.ComponentController.Element (UniversalDashboard)
08:07:54 [Info] Microsoft.AspNetCore.Authentication.Cookies.CookieAuthenticationHandler Cookies was not authenticated. Failure message: Ticket expired
08:07:54 [Info] Microsoft.AspNetCore.Authentication.Cookies.CookieAuthenticationHandler Cookies was not authenticated. Failure message: Ticket expired
08:07:54 [Info] Microsoft.AspNetCore.Authentication.WsFederation.WsFederationHandler WsFederation was not authenticated. Failure message: Ticket expired
08:07:54 [Info] Microsoft.AspNetCore.Authorization.DefaultAuthorizationService Authorization failed.
08:07:54 [Info] Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker Authorization failed for the request at filter 'Microsoft.AspNetCore.Mvc.Authorization.AuthorizeFilter'.
08:07:54 [Info] Microsoft.AspNetCore.Mvc.ChallengeResult Executing ChallengeResult with authentication schemes (Cookies, WsFederation).
08:07:54 [Info] Microsoft.AspNetCore.Authentication.Cookies.CookieAuthenticationHandler AuthenticationScheme: Cookies was challenged.
08:07:54 [Info] Microsoft.AspNetCore.Authentication.WsFederation.WsFederationHandler AuthenticationScheme: WsFederation was challenged.
08:07:54 [Info] Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker Executed action UniversalDashboard.Controllers.ComponentController.Element (UniversalDashboard) in 181.5415ms
08:07:54 [Info] Microsoft.AspNetCore.Hosting.Internal.WebHost Request finished in 230.4148ms 200 
08:07:54 [Info] Microsoft.AspNetCore.Hosting.Internal.WebHost Request starting HTTP/1.1 GET https://localhost:10002/favicon.ico 
```

**Comment by gerane** [comment]
This appears to be related to reauth and redirect. I just watched it happen. I was sitting on a page, and the page redirected to adfs auth page, and when it redirected back to the daskboard it was sitting at an internal api address.

**Comment by adamdriscoll** [comment]
@gerane FYI - Just reproduced this. I can get it fixed for 2.7.1. Thanks for your patience.

**Comment by adamdriscoll** [comment]
Resolved in 2.8.0 and tonight's nightly build.

**Comment by gerane** [comment]
@adamdriscoll I'll try to look at this nightly build sometime today or tonight.
